src/components/filesystem/index.js

```diff
@@ -1021,11 +1026,22 @@ module.exports = {
             var filename = file.getFileProperties().name;
             let latestFile = this.files.filter(f => f.getName() == filename)[0];
             this.selectedFiles = [latestFile];
-            var allSharedWithUsernames = this.getContext().sharedWith(latestFile);
-            var read_usernames = allSharedWithUsernames.left.toArray([]);
-            var edit_usernames = allSharedWithUsernames.right.toArray([]);
-            this.sharedWithData = {read_shared_with_users:read_usernames, edit_shared_with_users:edit_usernames};
-            this.showShare = true;
+            let that = this;
+            this.showSpinner = true;
+            this.updateSocial(function(result) {
```

Comment:
we need to avoid this and teh getbypath call. There should be no network requests to show share modal